sdk/handler.go

```diff
@@ -1,15 +1,54 @@
 package sdk
 
 import (
+	"context"
 	"crypto/tls"
 	"fmt"
+	"io"
 	"net"
 	"net/http"
 	"os"
+	"time"
 )
 
 const activatePath = "/Plugin.Activate"
 
+// serverCloser is an io.Closer whose Close method shuts down the referenced http.Server using its ShutDown method.
+// The channel 'err' receives the error produced by the Serve method of the server. This error is returned by the
+// Close method.
+type serverCloser struct {
+	server *http.Server
+	err    chan error
+}
+
+func (s serverCloser) Close() error {
+	// Wait at most 10 seconds for the server.Shutdown method to return
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
Maybe make the 10s configurable? 10s in some use-cases is quite long

Reply:
Makes sense, I will see about adding that somehow.